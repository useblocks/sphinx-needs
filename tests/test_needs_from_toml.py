import json
from pathlib import Path

import pytest
from syrupy.filters import props


@pytest.mark.parametrize(
    "test_app",
    [
        {
            "buildername": "html",
            "srcdir": "doc_test/needs_from_toml",
            "no_plantuml": True,
        }
    ],
    indirect=True,
)
def test_needs_from_toml(test_app, snapshot):
    app = test_app
    app.build()
    assert not app._warning.getvalue()
    data = json.loads(Path(app.outdir, "needs.json").read_text("utf8"))
    assert data == snapshot(
        exclude=props("created", "project", "creator", "needs_schema")
    )
