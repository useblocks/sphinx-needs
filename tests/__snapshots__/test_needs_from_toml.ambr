# serializer version: 1
# name: test_needs_from_toml[test_app0]
  dict({
    'current_version': '0.1.0',
    'versions': dict({
      '0.1.0': dict({
        'needs': dict({
          'O_001': dict({
            'custom': 'custom value',
            'docname': 'index',
            'external_css': 'external_link',
            'full_title': 'Test need',
            'id': 'O_001',
            'layout': '',
            'lineno': 4,
            'section_name': 'TEST DOCUMENT',
            'sections': list([
              'TEST DOCUMENT',
            ]),
            'status': 'open',
            'title': 'Test need',
            'type': 'other',
            'type_name': 'Another need type',
          }),
        }),
        'needs_amount': 1,
        'needs_defaults_removed': True,
      }),
    }),
  })
# ---
